File: Others/easy/MinimumWaitingTime.py

def minimumWaitingTime(queries):
    queries = sorted(queries)

    for i in range(1, len(queries) - 1):
        queries[i] += queries[i - 1]

    return sum(queries[:-1])


print(minimumWaitingTime([3, 2, 1, 2, 6]))

File: Golang/easy/minimum-waiting-time.go
package main

import "sort"

func MinimumWaitingTime(queries []int) int {
	sort.Ints(queries)
	sum := 0
	sumSoFar := 0

	for _, elem := range queries[:len(queries)-1] {
		sumSoFar += elem
		sum += sumSoFar
	}

	return sum
}
